src/platform/packages/private/kbn-esql-editor/src/editor_footer/esql_starred_queries_service.tsx

```diff
@@ -81,9 +82,13 @@ export class EsqlStarredQueriesService {
   static async initialize(services: EsqlStarredQueriesServices) {
     const client = new FavoritesClient<StarredQueryMetadata>('esql_editor', 'esql_query', {
       http: services.http,
+      userProfile: services.userProfile,
       usageCollection: services.usageCollection,
     });
 
+    const isAvailable = await client.isAvailable();
+    if (!isAvailable) return null;
```

Comment:
is it worth extending this conditional check here to the dashboard list implementation to determine if favorites is enabled here https://github.com/elastic/kibana/blob/v8.17.0/packages/content-management/table_list_view_table/src/services.tsx#L278, especially that when we navigate to the dashboard listing page at attempt is still made to request favorites?

Reply:
I think it would be ideal, but that check is async so I wanted to avoid additional changes. 
I will take another look

Reply:
I just changed that isFavoritesEnabled to allow for Promise

Reply:
Also fixes @azasypkin's suggestion: https://github.com/elastic/kibana/pull/204397#pullrequestreview-2513913909